include/swift/Reflection/TypeRef.h

```diff
@@ -733,25 +734,32 @@ class UnmanagedStorageTypeRef final : public ReferenceStorageTypeRef {
 };
 
 class SILBoxTypeRef final : public TypeRef {
-  const TypeRef *BoxedType;
+  llvm::PointerIntPair<const TypeRef *, 1, bool> BoxedTypeAndIsImmutable;
 
-  static TypeRefID Profile(const TypeRef *BoxedType) {
+  static TypeRefID Profile(const TypeRef *BoxedType, bool immutable) {
     TypeRefID ID;
     ID.addPointer(BoxedType);
+    ID.addInteger((uint32_t)immutable);
     return ID;
   }
 public:
-  SILBoxTypeRef(const TypeRef *BoxedType)
-    : TypeRef(TypeRefKind::SILBox), BoxedType(BoxedType) {}
+  SILBoxTypeRef(const TypeRef *BoxedType, bool immutable)
```

Comment:
Unless immutable boxes will get a different mangling you probably don't need to track this in the TypeRef